packages/lab/src/stepped-tracker/TrackerStep/TrackerStep.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  ComponentPropsWithoutRef,
+  CSSProperties,
+  forwardRef,
+  useEffect,
+} from "react";
+import { clsx } from "clsx";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { makePrefixer, Tooltip } from "@salt-ds/core";
+import {
+  StepActiveIcon,
+  StepDefaultIcon,
+  StepSuccessIcon,
+} from "@salt-ds/icons";
+import { TrackerConnector } from "../TrackerConnector";
+
+import {
+  useSteppedTrackerContext,
+  useTrackerStepContext,
+} from "../SteppedTracker";
+
+import trackerStepCss from "./TrackerStep.css";
+
+const withBaseName = makePrefixer("saltTrackerStep");
+
+type State = "default" | "completed";
+
+type StateWithActive = State | "active";
+
+export interface TrackerStepProps extends ComponentPropsWithoutRef<"li"> {
+  /**
+   * The state of the TrackerStep
+   */
+  state?: State;
+}
+
+const iconMap = {
+  default: StepDefaultIcon,
+  completed: StepSuccessIcon,
+};
+
+const getStateIcon = ({
+  isActive,
+  state,
+}: {
+  isActive: boolean;
+  state: State;
+}) => {
+  if (state === "default" && isActive) {
+    return StepActiveIcon;
+  }
+  return iconMap[state];
+};
+
+const getState = ({
+  isActive,
+  state,
+}: {
+  isActive: boolean;
+  state: State;
+}): StateWithActive => {
+  if (state === "default" && isActive) {
+    return "active";
+  }
+  return state;
+};
+
+const useCheckWithinSteppedTracker = (isWithinSteppedTracker: boolean) => {
+  useEffect(() => {
+    if (process.env.NODE_ENV !== "production") {
+      if (!isWithinSteppedTracker) {
+        console.error(
+          "The TrackerStep component must be placed within a SteppedTracker component"
+        );
+      }
+    }
+  }, [isWithinSteppedTracker]);
+};
+
+export const TrackerStep = forwardRef<HTMLLIElement, TrackerStepProps>(
+  function TrackerStep(props, ref?) {
+    const { state = "default", className, children, ...restProps } = props;
+
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-tracker-step",
+      css: trackerStepCss,
+      window: targetWindow,
+    });
+
+    const { activeStep, totalSteps, hasTooltips, isWithinSteppedTracker } =
+      useSteppedTrackerContext();
+    const stepNumber = useTrackerStepContext();
+
+    useCheckWithinSteppedTracker(isWithinSteppedTracker);
+
+    const isActive = activeStep === stepNumber;
+    const Icon = getStateIcon({ isActive, state });
+    const resolvedState = getState({ isActive, state });
+    const connectorState = activeStep > stepNumber ? "active" : "default";
+    const hasConnector = stepNumber < totalSteps - 1;
+
+    const Inner = (
+      <li
+        className={clsx(withBaseName(), withBaseName(resolvedState), className)}
+        style={
+          {
+            ...props.style,
```

Comment:
you could take the style out from restProps, I think otherwise they're being added again below?
I would take it out and make `const innerStyle = {...style, "--saltTrackerStep-width": `${100 / totalSteps}%`}` like we do for the other components

Reply:
Good spot! I had moved the spreading of ..rest after the other props in response to feedback where this wouldn't be an issue, but hadn't spotted to update this.